app/src/main/java/ru/yandex/practicum/contacts/presentation/main/ContactUi.java

```diff
@@ -41,6 +42,11 @@ public List<ContactType> getTypes() {
         return types;
     }
 
+    @Override
+    public boolean theSameAs(ContactUi contactUi) {
+        return false;
```

Comment:
Здесь нужно сравнить текущий объект с объектом из аргумента:
```suggestion
       return this.hashCode() == contactUi.hashCode();
```